massa-ledger/src/ledger_db.rs

```diff
@@ -0,0 +1,340 @@
+// Copyright (c) 2022 MASSA LABS <info@massa.net>
+
+//! Module to interact with the disk ledger
+
+use massa_hash::{Hash, HASH_SIZE_BYTES};
+use massa_models::{Address, Amount, DeserializeCompact, SerializeCompact};
+use rocksdb::{
+    ColumnFamilyDescriptor, Direction, IteratorMode, Options, ReadOptions, WriteBatch, DB,
+};
+use std::collections::BTreeMap;
+
+use crate::{ledger_changes::LedgerEntryUpdate, LedgerEntry, SetOrDelete, SetOrKeep};
+
+// TODO: remove rocks_db dir when sled is cut out
+const DB_PATH: &str = "../massa-node/storage/ledger/rocks_db";
+const LEDGER_CF: &str = "ledger";
+const METADATA_CF: &str = "metadata";
+const OPEN_ERROR: &str = "critical: rocksdb open operation failed";
+const CRUD_ERROR: &str = "critical: rocksdb crud operation failed";
+const CF_ERROR: &str = "critical: rocksdb column family operation failed";
+const BALANCE_IDENT: u8 = 0u8;
+const BYTECODE_IDENT: u8 = 1u8;
+
+/// Ledger sub entry enum
+pub enum LedgerSubEntry {
+    /// Balance
+    Balance,
+    /// Bytecode
+    Bytecode,
+    /// Datastore entry
+    Datastore(Hash),
+}
+
+/// Disk ledger DB module
+///
+/// Contains a RocksDB DB instance
+pub(crate) struct LedgerDB(pub(crate) DB);
+
+/// Destroy the disk ledger db and free the lock
+pub fn destroy_ledger_db() {
+    DB::destroy(&Options::default(), DB_PATH).expect(OPEN_ERROR);
+}
+
+/// Balance key formatting macro
+///
+/// NOTE: ident being in front of addr is required for the intermediate bootstrap implementation
+macro_rules! balance_key {
+    ($addr:ident) => {
+        [&[BALANCE_IDENT], &$addr.to_bytes()[..]].concat()
+    };
+}
+
+/// Bytecode key formatting macro
+///
+/// NOTE: ident being in front of addr is required for the intermediate bootstrap implementation
+/// NOTE: still handle separate bytecode for now to avoid too many refactoring at once
+macro_rules! bytecode_key {
+    ($addr:ident) => {
+        [&[BYTECODE_IDENT], &$addr.to_bytes()[..]].concat()
+    };
+}
+
+/// Datastore entry key formatting macro
+///
+/// TODO: add a separator identifier if the need comes to have multiple datastores
+macro_rules! data_key {
+    ($addr:ident, $key:ident) => {
+        [&$addr.to_bytes()[..], &$key.to_bytes()[..]].concat()
+    };
+}
+
+/// Datastore entry prefix formatting macro
+macro_rules! data_prefix {
+    ($addr:ident) => {
+        &$addr.to_bytes()[..]
+    };
+}
+
+/// Compute the end bound for a given prefix
+pub fn end_prefix(prefix: &[u8]) -> Option<Vec<u8>> {
```

Comment:
I am still confused by what this function does exactly :)

Reply:
Improved the comment.